src/c_api/prepared_statement.cpp

```diff
@@ -25,138 +25,268 @@ void kuzu_prepared_statement_destroy(kuzu_prepared_statement* prepared_statement
         delete static_cast<std::unordered_map<std::string, std::unique_ptr<Value>>*>(
             prepared_statement->_bound_values);
     }
-    free(prepared_statement);
 }
 
 bool kuzu_prepared_statement_is_success(kuzu_prepared_statement* prepared_statement) {
     return static_cast<PreparedStatement*>(prepared_statement->_prepared_statement)->isSuccess();
 }
 
-char* kuzu_prepared_statement_get_error_message(kuzu_prepared_statement* prepared_statement) {
-    auto error_message =
-        static_cast<PreparedStatement*>(prepared_statement->_prepared_statement)->getErrorMessage();
-    if (error_message.empty()) {
-        return nullptr;
+kuzu_state kuzu_prepared_statement_get_error_message(kuzu_prepared_statement* prepared_statement,
+    char** out_error_message) {
+    if (prepared_statement == nullptr || prepared_statement->_prepared_statement == nullptr) {
+        return KuzuError;
+    }
+    try {
+        auto error_message =
+            static_cast<PreparedStatement*>(prepared_statement->_prepared_statement)
+                ->getErrorMessage();
+        if (error_message.empty()) {
```

Comment:
The empty error message should not be treated as an error. I think it makes more sense to return it as-is.